src/thermo/PlasmaPhase.cpp

```diff
@@ -0,0 +1,236 @@
+}
+
+void PlasmaPhase::getParameters(AnyMap& phaseNode) const
+{
+    IdealGasPhase::getParameters(phaseNode);
+    AnyMap eedf;
+    eedf["type"] = m_distributionType;
+    vector_fp levels(m_nPoints);
+    Eigen::Map<Eigen::ArrayXd>(levels.data(), m_nPoints) = m_electronEnergyLevels;
+    eedf["energy-levels"] = levels;
+    if (m_distributionType == "isotropic") {
+        eedf["shape-factor"] = m_isotropicShapeFactor;
+        eedf["mean-electron-energy"].setQuantity(m_meanElectronEnergy, "eV");
+    } else if (m_distributionType == "discretized") {
+        vector_fp dist(m_nPoints);
+        Eigen::Map<Eigen::ArrayXd>(dist.data(), m_nPoints) = m_electronEnergyDist;
+        eedf["distribution"] = dist;
+    }
+    phaseNode["electron-energy-distribution"] = std::move(eedf);
+}
+
+void PlasmaPhase::setParameters(const AnyMap& phaseNode, const AnyMap& rootNode)
+{
+    IdealGasPhase::setParameters(phaseNode, rootNode);
+    if (phaseNode.hasKey("electron-energy-distribution")) {
+        const AnyMap eedf = phaseNode["electron-energy-distribution"].as<AnyMap>();
+        m_distributionType = eedf["type"].asString();
+        if (m_distributionType == "isotropic") {
+            if (eedf.hasKey("shape-factor")) {
+                m_isotropicShapeFactor = eedf["shape-factor"].asDouble();
+            } else {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "isotropic type requires shape-factor key.");
+            }
+            if (eedf.hasKey("mean-electron-energy")) {
+                double energy = eedf.convert("mean-electron-energy", "eV");
+                setMeanElectronEnergy(energy);
+            } else {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "isotropic type requires electron-temperature key.");
+            }
+        } else if (m_distributionType == "discretized") {
+            if (!eedf.hasKey("energy-levels")) {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "Cannot find key energy-levels.");
+            }
+            if (!eedf.hasKey("distribution")) {
+                throw CanteraError("PlasmaPhase::setParameters",
+                    "Cannot find key distribution.");
+            }
+            setElectronEnergyDistribution(eedf["energy-levels"].asVector<double>(),
+                                          eedf["distribution"].asVector<double>());
+        }
+    }
+}
+
+bool PlasmaPhase::addSpecies(shared_ptr<Species> spec)
+{
+    bool added = IdealGasPhase::addSpecies(spec);
+    size_t nElectronSpecies = 0;
+    for (size_t k = 0; k < nSpecies(); k++) {
+        if (molecularWeight(k) == atomicWeight(elementIndex("E"))) {
+            m_electronSpeciesIndex = k;
+            nElectronSpecies++;
+        }
+    }
+    if (nElectronSpecies == 0) {
+        throw CanteraError("PlasmaPhase::addSpecies",
+                           "No electron species found.");
```

Comment:
Since `addSpecies` is called for each species, this seems like it creates a requirement for the electron to be the first species in the phase, which I don't think we need to require.